worker/sandbox.go

```diff
@@ -1,60 +1,32 @@
 package worker
 
 import (
-	"os"
-	"path/filepath"
-	"time"
+	"log"
 
+	"github.com/natsukagami/kjudge/worker/sandbox"
+	"github.com/natsukagami/kjudge/worker/sandbox/isolate"
+	"github.com/natsukagami/kjudge/worker/sandbox/raw"
 	"github.com/pkg/errors"
 )
 
-// Sandbox provides a way to run an arbitary command
-// within a sandbox, with configured input/outputs and
-// proper time and memory limits.
-//
-// kjudge currently implements two sandboxes, "isolate" (which requires "github.com/ioi/isolate" to be available)
-// and "raw" (NOT RECOMMENDED, RUN AT YOUR OWN RISK).
-// Which sandbox is used can be set at runtime with a command-line switch.
-type Sandbox interface {
-	Run(*SandboxInput) (*SandboxOutput, error)
-}
-
-// SandboxInput is the input to a sandbox.
-type SandboxInput struct {
-	Command     string            `json:"command"`      // The passed command
-	Args        []string          `json:"args"`         // any additional arguments, if needed
-	Files       map[string][]byte `json:"files"`        // Any additional files needed
-	TimeLimit   time.Duration     `json:"time_limit"`   // The given time-limit
-	MemoryLimit int               `json:"memory_limit"` // in KBs
-
-	CompiledSource []byte `json:"compiled_source"` // Should be written down to the CWD as a file named "code", as the command expects
-	Input          []byte `json:"input"`
-}
-
-// SandboxOutput is the output which the sandbox needs to give back.
-type SandboxOutput struct {
-	Success     bool          `json:"success"`      // Whether the command exited zero.
-	RunningTime time.Duration `json:"running_time"` // The running time of the command.
-	MemoryUsed  int           `json:"memory_used"`  // in KBs
-
-	Stdout       []byte `json:"stdout"`
-	Stderr       []byte `json:"stderr"`
-	ErrorMessage string `json:"error_message,omitempty"`
-}
-
-// CopyTo copies all the files it contains into cwd.
-func (input *SandboxInput) CopyTo(cwd string) error {
-	// Copy all the files into "cwd"
-	for name, file := range input.Files {
-		if err := os.WriteFile(filepath.Join(cwd, name), file, 0666); err != nil {
-			return errors.Wrapf(err, "writing file %s", name)
-		}
+func NewSandbox(name string, ignoreWarnings ...bool) (sandbox.Runner, error) {
+	if len(ignoreWarnings) > 1 {
+		return nil, errors.New("Function NewSandbox only takes 1 or 2 arguments")
 	}
```

Comment:
Let's not do this. We have no uses for it (yet), the warning is useful (running arbitrary code on your computer always is), and this is not idiomatic Go.

In go you would either have a special function "overload" with an opt parameter
```go
func NewSandbox(name string) { ... }
func NewSandboxOpts(name string, opts *sandbox.Opts) { ... }
```
or use the "functional option" approach [like this](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis).

Reply:
The reasoning behind this change was that I've been writing performance tests and the warning will show up for every test. Anyway, the functional approach lgtm. I will implement it.

Reply:
Implemented. Thank you so much for the help.